components/input/navigationRectangle.js

```diff
@@ -1,130 +1,100 @@
+            fontSize: 20,
           }}
         >
-          Shot {shotIndex}/{attemptShots.length}
+          Shot <Text style={{ fontWeight: "bold" }}>{shot.shotNum}</Text>
         </Text>
       </View>
 
       <View
         style={{
-          flex: 1,
           flexDirection: "column",
           alignItems: "center",
-          justifyContent: "space-between",
+          justifyContent: "center",
         }}
       >
         <View
           style={{
-            marginBottom: 10,
-            borderBottomWidth: 2,
-            borderColor: "#A0A0A0",
+            flexDirection: "row",
+            justifyContent: "space-between",
+            gap: 10,
           }}
         >
-          {requirementDisplay(attemptShots, drillInfo, shotIndex)}
+          {drillInfo.requirements.map((requirement) => (
+            <Text
+              key={requirement.name}
+              style={{
+                fontSize: 16,
+                padding: 2,
+              }}
+            >
+              {requirement.prompt}:
+              <Text style={{ fontWeight: "bold" }}>
+                {" "}
+                {shot.items[requirement.name]} {requirement.distanceMeasure}
+              </Text>
+            </Text>
+          ))}
         </View>
 
+        {keys.length > 0 && !keys.some((key) => !inputValues[key]) && (
+          <Divider
+            style={{ backgroundColor: "#A0A0A0", height: 1, width: "100%" }}
+          />
+        )}
+
         <View
           style={{
             flexDirection: "row",
-            marginBottom: 10,
-            alignItems: "center",
+            justifyContent: "space-evenly",
           }}
         >
-          {drillInfo.inputs.map((item, id) => (
-            <View
-              style={{
-                flexDirection: "row",
-                alignItems: "center",
-                justifyContent: "center",
-                margin: 10,
-                flex: 1,
-              }}
-              key={id}
-            >
-              <Icon source={getIconByKey(item.id)} size={15} />
-              <Text style={{ fontSize: 13, padding: 2 }}>
-                {inputValues[item.id]} {item.distanceMeasure}
-              </Text>
-            </View>
-          ))}
+          {drillInfo.inputs.map((input, id) => {
+            if (inputValues[input.id]) {
+              return (
+                <View
+                  style={{
+                    flexDirection: "row",
+                    alignItems: "center",
+                    justifyContent: "center",
+                    marginTop: 10,
+                  }}
+                  key={id}
+                >
+                  <Icon source={getIconByKey(input.id)} size={15} />
+                  <Text style={{ fontSize: 13, padding: 2 }}>
+                    {inputValues[input.id]} {input.distanceMeasure}
+                  </Text>
+                </View>
+              );
+            }
+          })}
         </View>
```

Comment:
More of just a compliment here, I like how you have it not displaying anything if there is no data. Although, one possible idea is having it return text that says "current shot" if this is the case or we can leave it blank. Just a small thought

Reply:
Hm, current shot does sound like a good idea. Let's merge this and you can try to do that on your PR.